runtime/src/assets.rs

```diff
@@ -67,9 +71,70 @@ impl AssetRegistry {
     }
 }
 
+lazy_static! {
+    // NOTE: restrict access to the lock to ensure that no async code yields while holding the mutex
+    static ref LENDING_ASSETS: Mutex<LendingAssets> = Mutex::new(LendingAssets::default());
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct LendingAssets {
+    underlying_to_lend_token: BTreeMap<CurrencyId, CurrencyId>,
+    lend_token_to_underlying: BTreeMap<CurrencyId, CurrencyId>,
+}
+
+impl LendingAssets {
+    /// Fetch the global, mutable singleton
+    fn global() -> Result<MutexGuard<'static, Self>, Error> {
+        LENDING_ASSETS.lock().map_err(|_| Error::CannotOpenAssetRegistry)
+    }
+
+    pub(crate) fn insert(underlying_id: CurrencyId, lend_token_id: CurrencyId) -> Result<(), Error> {
+        log::info!(
+            "Found loans market: {:?}, with lend token: {:?}",
```

Comment:
FYI this prints the full struct rather than the ticker, because the concurrent listener might run before the foreign asset listener